### PR TITLE
chore: add basic gh workflows

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
**Description**

Stumbled over Min.io  and this plugin today whilst researching dependency caching and noticed that this repository was lacking some default GitHub workflows.
I thought why not open a PR to add the workflows found in many of the new Jenkins plugin repositories.
Feel free to give me feedback, and I shall implement it as soon as I can or close this PR if you think this is not what you want to do.
Either way, nice little plugin :+1: !

**Changes**

* add build badge to readme
* add dependabbot config
* add CodeQL GitHub Action workflow
* add Release Drafter config with GitHub Action workflow (this workflow should automatically creates a GitHub release once a mvn release commit is detected)
* add a GitHub Action workflow to build and upload a .hpi file to a new GH release

**Issues**

* n/a

### CODE REVIEW

**Comment by h1dden-da3m0n** [comment]
Sorry for the ping @jenkinsci/minio-plugin-developers, but may I ask for some feedback on this?

**Comment by Sleepy-GH** [comment]
> Sorry for the ping @jenkinsci/minio-plugin-developers, but may I ask for some feedback on this?

I'm so sorry, I never got a notification that this PR was created! 

It looks like a cool addition, thanks for taking the time to create this! Do you have an example repository that also makes use of github actions? I'll take a closer look after Christmas. 

Happy holidays!

**Comment by h1dden-da3m0n** [comment]
Hi Sleepy-GH,

no worries, just thought I might send a ping b4 I forget too :sweat_smile:
Sure thing:
* [Rest-List-Param](https://github.com/jenkinsci/rest-list-parameter-plugin) (plugging my plugin :sweat_smile:) 
* [JenkinsCi .github repo with docu and info regarding releas drafter](https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.adoc)
* [custom-folder-icon-plugin](https://github.com/jenkinsci/custom-folder-icon-plugin)
* [image-tag-param](https://github.com/jenkinsci/image-tag-parameter-plugin)

Take all the time u need to review and any feedback, questions are welcome and will be integrated as soon as I can.
Merry Christmas and happy holidays to you too!

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Inherit base settings from https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml
+_extends: .github
+name-template: REST List Parameter v$RESOLVED_VERSION
```
Comment (Sleepy-GH) [inline .github/release-drafter.yml:3]: This should probably have a different name. The tag-template as well

File: .github/workflows/upload-release-artifact.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Upload release artifact
+
+on:
+  release:
+    types: [ published ]
+
+env:
+  PROJECT: rest-list-parameter
```
Comment (Sleepy-GH) [inline .github/workflows/upload-release-artifact.yml:8]: This should also have a different name

**Comment by Sleepy-GH** [comment]
Other then some copy-paste things that should be changed, everything else looks cool!

File: .github/release-drafter.yml (hunk 1 above)
Comment (h1dden-da3m0n) [reply .github/release-drafter.yml:3]: All I can say in my defense ... :facepalm:  I am sorry for this f up.

File: .github/workflows/upload-release-artifact.yml (hunk 2 above)
Comment (h1dden-da3m0n) [reply .github/workflows/upload-release-artifact.yml:8]: same as above, sorry! (note to self: stop copying things everywhere without proper second pass of updating the values)

**Comment by h1dden-da3m0n** [comment]
Okay, fixed!

File: .github/release-drafter.yml (hunk 1 above)
Comment (Sleepy-GH) [reply .github/release-drafter.yml]: It's all good. If I got a nickel for every time I made a copy paste mistake I'd be retired by now 😉

**Comment by Sleepy-GH** [comment]
Thanks, I'll go ahead and merge this and we'll see what updates dependabot has in store!

**Pull request merged**